src/time_integration/paired_explicit_runge_kutta/methods_PERK3.jl

```diff
@@ -0,0 +1,368 @@
+                                                    verbose)
+    end
+    # Fill A-matrix in P-ERK style
+    a_matrix = zeros(num_stages - 2, 2)
+    a_matrix[:, 1] = c[3:end]
+    a_matrix[:, 1] -= a_unknown
+    a_matrix[:, 2] = a_unknown
+
+    return a_matrix, c, dt_opt
+end
+
+# Compute the Butcher tableau for a paired explicit Runge-Kutta method order 3
+# using provided values of coefficients a in A-matrix of Butcher tableau
+function compute_PairedExplicitRK3_butcher_tableau(num_stages,
+                                                   base_path_a_coeffs::AbstractString;
+                                                   cS2)
+
+    # Initialize array of c
+    c = compute_c_coeffs(num_stages, cS2)
+
+    # - 2 Since First entry of A is always zero (explicit method) and second is given by c_2 (consistency)
+    a_coeffs_max = num_stages - 2
+
+    a_matrix = zeros(a_coeffs_max, 2)
+    a_matrix[:, 1] = c[3:end]
+
+    path_a_coeffs = joinpath(base_path_a_coeffs,
+                             "a_" * string(num_stages) * ".txt")
+
+    @assert isfile(path_a_coeffs) "Couldn't find file $path_a_coeffs"
+    a_coeffs = readdlm(path_a_coeffs, Float64)
+    num_a_coeffs = size(a_coeffs, 1)
+
+    @assert num_a_coeffs == a_coeffs_max
+    # Fill A-matrix in P-ERK style
+    a_matrix[:, 1] -= a_coeffs
+    a_matrix[:, 2] = a_coeffs
+
+    return a_matrix, c
+end
+
+@doc raw"""
+    PairedExplicitRK3(num_stages, base_path_a_coeffs::AbstractString, dt_opt;
+                      cS2 = 1.0f0)
+    PairedExplicitRK3(num_stages, tspan, semi::AbstractSemidiscretization;
+                      verbose = false, cS2 = 1.0f0)
+    PairedExplicitRK3(num_stages, tspan, eig_vals::Vector{ComplexF64};
+                      verbose = false, cS2 = 1.0f0)
+
+    Parameters:
+    - `num_stages` (`Int`): Number of stages in the paired explicit Runge-Kutta (P-ERK) method.
+    - `base_path_a_coeffs` (`AbstractString`): Path to a file containing some coefficients in the A-matrix in 
+      the Butcher tableau of the Runge Kutta method.
+      The matrix should be stored in a text file at `joinpath(base_path_a_coeffs, "a_$(num_stages).txt")` and separated by line breaks.
+    - `dt_opt` (`Float64`): Optimal time step size for the simulation setup.
+    - `tspan`: Time span of the simulation.
+    - `semi` (`AbstractSemidiscretization`): Semidiscretization setup.
+    -  `eig_vals` (`Vector{ComplexF64}`): Eigenvalues of the Jacobian of the right-hand side (rhs) of the ODEProblem after the
+      equation has been semidiscretized.
+    - `verbose` (`Bool`, optional): Verbosity flag, default is false.
+    - `cS2` (`Float64`, optional): Value of c in the Butcher tableau at c_{s-2}, when
+      s is the number of stages, default is 1.0f0.
+
+The following structures and methods provide an implementation of
+the third-order paired explicit Runge-Kutta (P-ERK) method
+optimized for a certain simulation setup (PDE, IC & BC, Riemann Solver, DG Solver).
```

Comment:
Could you please clarify in the docstring that this relies on the extensions for Convex and ECOS as well as on the extension for NLSolve, i.e. that one has to import all three of them in order to use `PairedExplicitRK3` (at least if one doesn't pass the `base_path_a_coeffs` as far as I understand)? Correct me if that's wrong. I see that no such clarification is in the docstring for `PairedExplicitRK2` regarding the Convex + ECOS extension. Could you also add that there, please (either here or in another PR)?

Reply:
I will! Thank you for the suggestion! I did this with PERK2 in #2123 .